external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutRetryExponentialBackoff.java

```diff
@@ -172,19 +174,20 @@ public KafkaSpoutRetryExponentialBackoff(TimeInterval initialDelay, TimeInterval
     }
 
     @Override
-    public Set<TopicPartition> retriableTopicPartitions() {
-        final Set<TopicPartition> tps = new HashSet<>();
+    public Map<TopicPartition, Long> earliestOffsetForEachRetriableTopicPartition() {
```

Comment:
Let's rename this method to `earliestRetriableOffsets`. The javadoc will provide the details. Let's not write very complex and long method names.